openedx_learning/apps/authoring/publishing/api.py

```diff
@@ -449,13 +465,199 @@ def set_draft_version(
     edited some content). Setting a Draft's version to None is like deleting it
     from Studio's editing point of view (see ``soft_delete_draft`` for more
     details).
+
+    Calling this function attaches a new DraftChange and attaches it to a
+    DraftChangeLog.
+
+    This function will create DraftSideEffect entries and properly add any
+    containers that may have been affected by this draft update, UNLESS it is
+    called from within a bulk_draft_changes_for block. If it is called from
+    inside a bulk_draft_changes_for block, it will not add side-effects for
+    containers, as bulk_draft_changes_for will automatically do that when the
+    block exits.
+    """
+    if set_at is None:
+        set_at = datetime.now(tz=timezone.utc)
+    
+    tx_context = atomic() if create_transaction else nullcontext()
```

Comment:
New convention: optionally creating a transaction, based on passed in parameters. This is to avoid unnecessarily opening transactions when we're being invoked from things that already open transactions.